### PR TITLE
RUM-4148 Add support to the Toolbar in Session Replay

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

This PR allows sessin replay to capture the content of a Toolbar. 

- it let's the default mapping handle all the toolbar's children
- it capture's the `ActionBarContainer`'s background

### CODE REVIEW

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/DefaultRecorderProvider.kt [hunk 1]
```diff
@@ -17,14 +17,15 @@ import android.widget.NumberPicker
 import android.widget.RadioButton
 import android.widget.SeekBar
 import android.widget.TextView
+import androidx.appcompat.widget.ActionBarContainer
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/DefaultRecorderProvider.kt:20]: interesting enough there is also [android.internal.widget.ActionBarContainer ](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/com/android/internal/widget/ActionBarContainer.java), should we support it?

File: features/dd-sdk-android-session-replay/src/main/kotlin/androidx/appcompat/widget/ActionBarContainerDrawableAccessor.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package androidx.appcompat.widget
+
+import android.annotation.SuppressLint
+import android.graphics.drawable.Drawable
+
+internal class ActionBarContainerDrawableAccessor(
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/androidx/appcompat/widget/ActionBarContainerDrawableAccessor.kt:13]: minor: maybe add `Datadog` to the name? unlikely that the class named `ActionBarContainerDrawableAccessor` will ever appear in Android codebase, but who knows.

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    override fun map(
+        view: ActionBarContainer,
+        mappingContext: MappingContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
+    ): List<MobileSegment.Wireframe> {
+        // The ActionBarContainer uses an internal Drawable implementation that redirects to some fields in the
+        // ActionBarContainer class.
+        // Fortunately, the mBackground field we're interested in is package private,
+        // which allows us to access it.
+//        val background = backgroundField?.get(view) as? Drawable
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt:40]: leftover?

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        val background = ActionBarContainerDrawableAccessor(view).getBackgroundDrawable()
+        val shapeStyle = background?.let { resolveShapeStyle(it, view.alpha) }
+        val id = viewIdentifierResolver.resolveChildUniqueIdentifier(view, PREFIX_BACKGROUND_DRAWABLE)
+
+        if ((shapeStyle != null) && (id != null)) {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt:45]: minor: extra brackets

File: features/dd-sdk-android-session-replay/src/main/kotlin/androidx/appcompat/widget/ActionBarContainerDrawableAccessor.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    val container: ActionBarContainer
+) {
+
+    @SuppressLint("RestrictedApi")
+    fun getBackgroundDrawable(): Drawable? {
+        return container.mBackground
+    }
+
+    @SuppressLint("RestrictedApi")
+    fun setBackgroundDrawabal(drawable: Drawable) {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/androidx/appcompat/widget/ActionBarContainerDrawableAccessor.kt:23]: typo in the method name

```suggestion
    fun setBackgroundDrawable(drawable: Drawable) {
```

**Review by 0xnm — APPROVED** [review]

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/DefaultRecorderProvider.kt (hunk 1 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/DefaultRecorderProvider.kt:20]: That one is internal and not visible from customer applications. I'm not sure anyone uses it today, so for now let's say no. Anyway we have some telemetry RFC to list "unsupported" views to know which one we should support.

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt (hunk 3 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ActionBarContainerMapper.kt:40]: Yep

**Pull request merged**